source/LocationFilter.cpp

```diff
@@ -393,9 +393,11 @@ bool LocationFilter::Matches(const Ship &ship) const
 	{
 		// Create a set from the positive-valued attributes of this ship.
 		set<string> shipAttributes;
-		for(const auto &attr : ship.Attributes().Attributes())
-			if(attr.second > 0.)
-				shipAttributes.insert(shipAttributes.end(), attr.first);
+		ship.Attributes().Attributes().ForEach([&shipAttributes](const tuple<std::string, Attribute *, double> &attr)
+		{
+			if(get<2>(attr) > 0.)
```

Comment:
I think we try not to use tuples in general, pair already is limited to cases where it's simple like <thing, number>

that said I'd say its fine here given we always have the function above but we could always have a class for it? granted we already have a few classes added here